### PR TITLE
Replace pytest-freezegun with pytest-freezer

### PR DESCRIPTION
**Pull request by rebkwok**
pytest-freezegun is no longer actively maintained and has been moved into the main pytest-dev org as pytest-freezer (it's currently fine, but will break in python 3.12)

https://github.com/pytest-dev/pytest-freezer

### CODE REVIEW

File: requirements.dev.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.10
+# This file is autogenerated by pip-compile with Python 3.12
```
Comment (lucyb) [inline requirements.dev.txt:2]: This should be Python 3.10 as Reports hasn't been updated yet.

Comment (lucyb) [reply requirements.dev.txt:2]: Actually, nevermind me. It has been upgraded, but the .python-version is incorrect.

**Review by lucyb — APPROVED** [review]

File: requirements.dev.txt (hunk 1 above)
Comment (rebkwok) [reply requirements.dev.txt:2]: Ah, I saw the version had changed, and i checked the justfile's default python, and that the prod equivalent was generated with 3.12 and didn't look any further

Comment (rebkwok) [reply requirements.dev.txt:2]: I'll leave it for the rex team to sort out the .python-version :)

**Pull request merged**